src/berry_pots_app_tasks.c

```diff
@@ -0,0 +1,444 @@
+    int unk = ov17_02203CC8(data);
+    if (unk < 0) {
+        return 2;
+    }
+
+    *a1 = TRUE;
+    data->unk74 = 1;
+    return ov17_02203A74(data, unk);
+}
+
+static BOOL ov17_02203B88(BerryPotsAppData *data) {
+    BOOL unk = FALSE;
+
+    if (data->unk81 != 0xFF) {
+        return FALSE;
+    }
+
+    if (gSystem.newKeys & (PAD_BUTTON_Y | PAD_BUTTON_X | PAD_KEY_DOWN | PAD_KEY_UP | PAD_KEY_LEFT | PAD_KEY_RIGHT | PAD_BUTTON_B | PAD_BUTTON_A)) {
+        data->unk74 = 0;
+    }
+
+    if (gSystem.newKeys & PAD_KEY_LEFT) {
+        data->unk7C = (data->unk7C + (MAX_BERRY_POT - 1)) % MAX_BERRY_POT;
+        unk = TRUE;
+    } else if (gSystem.newKeys & PAD_KEY_RIGHT) {
+        data->unk7C = (data->unk7C + 1) % MAX_BERRY_POT;
+        unk = TRUE;
+    }
+
+    if (unk) {
+        data->unk81 = data->unk7C;
+        Set2dSpriteAnimActiveFlag(data->sprites[1], FALSE);
+        ov17_022028B8(data);
+        return TRUE;
+    }
+
+    return FALSE;
+}
+
+static BOOL ov17_02203C20(BerryPotsAppData *data, BOOL *a1) {
+    if (data->unk81 != 0xFF) {
+        return FALSE;
+    }
+
+    int unk = ov17_02203C78(data);
+    if (unk < 0 || data->unk7C == unk) {
+        return FALSE;
+    }
+
+    *a1 = TRUE;
+    data->unk74 = 1;
+    data->unk7C = unk;
+    data->unk81 = data->unk7C;
+    Set2dSpriteAnimActiveFlag(data->sprites[1], FALSE);
+    ov17_022028B8(data);
+
+    return TRUE;
+}
+
+static int ov17_02203C78(BerryPotsAppData *data) {
+    TouchscreenHitbox touchHitbox;
+
+    for (int i = 0, xOffsetMaybe = 16; i < MAX_BERRY_POT; xOffsetMaybe += 28, i++) {
+        touchHitbox.rect.left = xOffsetMaybe;
+        touchHitbox.rect.right = touchHitbox.rect.left + 22;
+        if (data->unk20[i].growthStage < BERRY_POT_GROWTH_STAGE_GROWING) {
+            touchHitbox.rect.top = 88;
+            touchHitbox.rect.bottom = touchHitbox.rect.top + 24;
+        } else {
+            touchHitbox.rect.top = 72;
+            touchHitbox.rect.bottom = touchHitbox.rect.top + 32;
+        }
+
+        if (TouchscreenHitbox_TouchNewIsIn(&touchHitbox)) {
+            return i;
+        }
+    }
+    return -1;
+}
+
+static int ov17_02203CC8(BerryPotsAppData *data) {
+    if (TouchscreenHitbox_TouchNewIsIn(&ov17_02203FC0)) {
+        return 5;
+    }
+
+    int unk = ov17_02203C78(data);
+    if (unk >= 0) {
+        data->unk7C = unk;
+        ov17_022028B8(data);
+        return 4;
+    }
+
+    return -1;
+}
+
+static u32 ov17_02203D00(BerryPotsAppData *data) {
+    BerryPotsAppData_UnkSub20 *unk = &data->unk20[data->unk7C];
+    switch (unk->growthStage) {
+        case BERRY_POT_GROWTH_STAGE_NONE:
+            return (unk->mulch != MULCH_NONE) ? 4 : 3;
+        case BERRY_POT_GROWTH_STAGE_PLANTED:
+        case BERRY_POT_GROWTH_STAGE_SPROUTED:
+        case BERRY_POT_GROWTH_STAGE_GROWING:
+        case BERRY_POT_GROWTH_STAGE_BLOOM:
+            return 5;
+        case BERRY_POT_GROWTH_STAGE_BERRIES:
+            return 6;
+
+        case 0xFF:
```

Comment:
Make a const for BERRY_POT_GROWTH_STAGE_NONE?

Reply:
1) that name is already used
2) can't find where growthState can be set to 0xFF